src/hypertable.c

```diff
@@ -2560,6 +2563,7 @@ ts_hypertable_osm_range_update(PG_FUNCTION_ARGS)
 	bool overlap = false, range_invalid = false;
 
 	/* Lock tuple FOR UPDATE */
+	Assert(!RecoveryInProgress());
```

Comment:
The assert cannot be used as a check in production deployments and is no guarantee that it will never be reached.

What happen if we run this code on a secondary that is in recovery? Will that cause a production issue?

If so, we should have a check in place so that the tuple is not locked when the server is in recovery, but lock it otherwise.

Reply:
Thanks for the review! I changed the commit message and added a comment.

> The assert cannot be used as a check in production deployments and is no guarantee that it will never be reached.

It wasn't meant as a runtime check, rather a documentation for myself and other devs. (Production images are not compiled with asserts enabled, right?). I added a comment there for clarity. This function would not normally be reached from the osm code: on replica it would fail before this call. But even when called directly from psql, it would not cause crash, it fail with the same error as [in the original issue](https://github.com/timescale/Support-Dev-Collab/issues/1847):
```sql
ERROR:  cannot assign TransactionIds during recovery
```
when trying to update catalog.

I can remove the assert altogether though. I only added it because this func calls `ts_chunk_get_osm_slice_and_lock` and it should not happen in recovery mode, so that we could catch it early if it happens.

Reply:
> Thanks for the review! I changed the commit message and added a comment.
> 
> > The assert cannot be used as a check in production deployments and is no guarantee that it will never be reached.
> 
> It wasn't meant as a runtime check, rather a documentation for myself and other devs. (Production images are not compiled with asserts enabled, right?). I added a comment there for clarity. This function would not normally be reached from the osm code: on replica it would fail before this call. But even when called directly from psql, it would not cause crash, it fail with the same error as [in the original issue](https://github.com/timescale/Support-Dev-Collab/issues/1847):
> 
> ```sql
> ERROR:  cannot assign TransactionIds during recovery
> ```
> 
> when trying to update catalog.

Thank you for explaining it.

> I can remove the assert altogether though. I only added it because this func calls `ts_chunk_get_osm_slice_and_lock` and it should not happen in recovery mode, so that we could catch it early if it happens.

No, keeping asserts is good for documentation purposes and declaring intentions and help debugging, as you mentioned above. I was just unsure if this meant that the code after would execute in production builds, causing havoc. If that cannot happen, a comment on why this cannot happen and a test (if possible) is good enough to ensure that we do not try to execute bad code in production builds.